### ISSUE TITLE
Limit on the number of agents per sweep?

### ISSUE THREAD

**Issue by pietrolesci** [wandb/client#1785]
**Describe the bug**
After initializing a sweep, running more than 3 agents causes an error

**To Reproduce**
Set up a sweep and run more than 3 agents

**Expected behavior**
No limit on the number of agents

**Screenshots**

**Desktop (please complete the following information):**
 - OS: tried on both own server running Ubuntu 18.04 and Google Colab
 - Browser: Chrome

**Additional context**
```
wandb: Starting wandb agent ??
2021-02-01 15:18:35,955 - wandb.wandb_agent - INFO - Running runs: []
400 response executing GraphQL.
{"errors":[{"message":"ValueError: '-4' is not in list","path":["agentHeartbeat"]}],"data":{"agentHeartbeat":null}}
wandb: ERROR Error while calling W&B API: ValueError: '-4' is not in list (<Response [400]>)
wandb: Terminating and syncing runs. Press ctrl-c to kill.
Traceback (most recent call last):
  File "/home/lesci/miniconda3/envs/deep_al/lib/python3.8/site-packages/wandb/sdk/internal/internal_api.py", line 1316, in agent_heartbeat
    response = self.gql(
  File "/home/lesci/miniconda3/envs/deep_al/lib/python3.8/site-packages/wandb/old/retry.py", line 96, in __call__
    result = self._call_fn(*args, **kwargs)
  File "/home/lesci/miniconda3/envs/deep_al/lib/python3.8/site-packages/wandb/sdk/internal/internal_api.py", line 137, in execute
    six.reraise(*sys.exc_info())
  File "/home/lesci/miniconda3/envs/deep_al/lib/python3.8/site-packages/six.py", line 703, in reraise
    raise value
  File "/home/lesci/miniconda3/envs/deep_al/lib/python3.8/site-packages/wandb/sdk/internal/internal_api.py", line 131, in execute
    return self.client.execute(*args, **kwargs)
  File "/home/lesci/miniconda3/envs/deep_al/lib/python3.8/site-packages/wandb/vendor/gql-0.2.0/gql/client.py", line 52, in execute
    result = self._get_result(document, *args, **kwargs)
  File "/home/lesci/miniconda3/envs/deep_al/lib/python3.8/site-packages/wandb/vendor/gql-0.2.0/gql/client.py", line 60, in _get_result
    return self.transport.execute(document, *args, **kwargs)
  File "/home/lesci/miniconda3/envs/deep_al/lib/python3.8/site-packages/wandb/vendor/gql-0.2.0/gql/transport/requests.py", line 39, in execute
    request.raise_for_status()
  File "/home/lesci/miniconda3/envs/deep_al/lib/python3.8/site-packages/requests/models.py", line 943, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: https://api.wandb.ai/graphql

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/lesci/miniconda3/envs/deep_al/bin/wandb", line 8, in <module>
    sys.exit(cli())
  File "/home/lesci/miniconda3/envs/deep_al/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/home/lesci/miniconda3/envs/deep_al/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/home/lesci/miniconda3/envs/deep_al/lib/python3.8/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/lesci/miniconda3/envs/deep_al/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/lesci/miniconda3/envs/deep_al/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/home/lesci/miniconda3/envs/deep_al/lib/python3.8/site-packages/click/decorators.py", line 21, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/lesci/miniconda3/envs/deep_al/lib/python3.8/site-packages/wandb/cli/cli.py", line 94, in wrapper
    return func(*args, **kwargs)
  File "/home/lesci/miniconda3/envs/deep_al/lib/python3.8/site-packages/wandb/cli/cli.py", line 815, in agent
    wandb_agent.agent(sweep_id, entity=entity, project=project, count=count)
  File "/home/lesci/miniconda3/envs/deep_al/lib/python3.8/site-packages/wandb/wandb_agent.py", line 562, in agent
    return run_agent(
  File "/home/lesci/miniconda3/envs/deep_al/lib/python3.8/site-packages/wandb/wandb_agent.py", line 519, in run_agent
    agent.run()
  File "/home/lesci/miniconda3/envs/deep_al/lib/python3.8/site-packages/wandb/wandb_agent.py", line 252, in run
    commands = self._api.agent_heartbeat(agent_id, {}, run_status)
  File "/home/lesci/miniconda3/envs/deep_al/lib/python3.8/site-packages/wandb/apis/internal.py", line 92, in agent_heartbeat
    return self.api.agent_heartbeat(*args, **kwargs)
  File "/home/lesci/miniconda3/envs/deep_al/lib/python3.8/site-packages/wandb/sdk/internal/internal_api.py", line 1327, in agent_heartbeat
    message = ast.literal_eval(e.args[0])["message"]
  File "/home/lesci/miniconda3/envs/deep_al/lib/python3.8/ast.py", line 59, in literal_eval
    node_or_string = parse(node_or_string, mode='eval')
  File "/home/lesci/miniconda3/envs/deep_al/lib/python3.8/ast.py", line 47, in parse
    return compile(source, filename, mode, flags,
  File "<unknown>", line 1
    400 Client Error: Bad Request for url: https://api.wandb.ai/graphql
        ^
SyntaxError: invalid syntax
```

**Comment by vanpelt** [comment]
@pietrolesci this looks like an issue with your sweep configuration: `ValueError: '-4' is not in list`.  Can you share either a link to your sweep or the sweep config?

**Comment by pietrolesci** [comment]
Hi @vanpelt, thanks a lot for your swift reply. Happy to share the link to the sweep, but it's a private project. How can I best handle this? In the meantime, here is the config file

```
program: sweep.py
method: bayes
metric:
  name: valid/loss_step
  goal: minimize
parameters:
  seed: 
    values: [42, 1994, 8975, 5713, 2917, 2620, 2020, 128, 1157]
  model_name:
    values: ["bert-base-uncased"]    
  batch_size:
    values: [8, 16, 32] 
  lr:
    values: [0.00025, 0.0005, 0.000025, 0.00005]
  num_epochs:
    values: [5, 10, 20]
  use_scheduler:
    values: [True, False]
  freeze_layers:
    values: [None, feature_extractor, -1, -2, -4]

early_terminate:
  type: hyperband
  s: 2
  eta: 3
  max_iter: 27
```

**Comment by raubitsj** [comment]
Sweep URL has been added to the internal ticket:
https://wandb.atlassian.net/browse/CLI-682

**Comment by pietrolesci** [comment]
Hi @vanpelt, I'd like to provide further feedback. I DO think it is related to the config because I managed to instantiate >4 agents running the same sweep by simply dropping that config (i.e. "freeze_layers"). However, I previously managed to successfully run experiments using the "-4" option (present in the sweep runs). I saw the same error with all the option "-1" too that, again, I managed to run. Once again, thank you very much for your help on this.
